### PR TITLE
7055 My_talks

### PR DESCRIPTION
**Pull request by ScreamCat1**

### CODE REVIEW

File: conference/react-js/src/reducers/talks.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const talksInitial = [];
 
 const talks = (state = talksInitial, action) => {
   if (action.type === LOAD) {
-    return [...state, ...action.payload];
+    return [...action.payload];
```
Comment (SupraVlad) [inline conference/react-js/src/reducers/talks.js]: seems like you can return `action.payload` by itself. I see no sense of using `spread` operator along with square brackets

File: conference/react-js/src/reducers/userTalks.js [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+  } else if (type === SORT_DATA) {
+    const { direction, talks, field } = payload;
+    const sortDirection = (a, b) => {
+      if (direction === ASC) {
+        return a[field] > b[field];
+      }
+      return a[field] < b[field];
+    };
+
+    return [...talks.sort(sortDirection)];
```
Comment (SupraVlad) [inline conference/react-js/src/reducers/userTalks.js]: same point about usage of `spread` + square brackets

**Review by SupraVlad — CHANGES_REQUESTED** [review]

**Review by SupraVlad — APPROVED** [review]

File: conference/react-js/src/components/Autorized-user-menu/index.jsx [hunk 3]
```diff
@@ -1,41 +1,38 @@
...
+        exact
+        className="tabs-list__anchor"
+        to={upcoming}
+        activeClassName="tabs-list__anchor_active"
+      >
+        Upcoming
+      </NavLink>
+    </li>
+    {length > 0 ?
+      (<li className="tabs-list__item">
```
Comment (asanametove) [inline conference/react-js/src/components/Autorized-user-menu/index.jsx]: I think you can write "length && <li ..." there

File: conference/react-js/src/containers/Talks/index.jsx [hunk 4]
```diff
@@ -57,12 +64,22 @@ class Talks extends Component {
     this.props.load(APPLY_FILTERS, { filter, listOfTalks });
   };
 
+  sortTalks = ({ target: { tagName, dataset: { name } } }) => {
+    if (tagName === 'TH' && name !== undefined) {
```
Comment (asanametove) [inline conference/react-js/src/containers/Talks/index.jsx]: Maybe "tagName === 'TH' && name" will be enough?

File: conference/react-js/src/reducers/talks.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ const talksInitial = [];
 
 const talks = (state = talksInitial, action) => {
```
Comment (asanametove) [inline conference/react-js/src/reducers/talks.js]: "const talks = (state = talksInitial, { type, payload })" looks better I think

File: conference/react-js/src/components/Autorized-user-menu/index.jsx (hunk 3 above)
Comment (ScreamCat1) [reply conference/react-js/src/components/Autorized-user-menu/index.jsx]: написал length > 0 && (<li...

**Review by asanametove — APPROVED** [review]

**Pull request merged**